modules/kit-elements-defaults/controller.php

```diff
@@ -0,0 +1,242 @@
+<?php
+namespace Elementor\Modules\KitElementsDefaults;
+
+use Elementor\Element_Base;
+use Elementor\Plugin;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Controller {
+
+	const NAMESPACE = 'elementor/v1';
+
+	const BASE_URL = '/kit-elements-defaults';
+
+	const META_KEY = '_elementor_elements_defaults_values';
+
+	public function register() {
+		add_action( 'rest_api_init', function () {
+			$this->register_index();
+
+			$this->register_update();
+
+			$this->register_destroy();
+		} );
+	}
+
+
+	/**
+	 * GET '/kit-elements-defaults'
+	 */
+	private function register_index() {
+		register_rest_route( static::NAMESPACE, static::BASE_URL, [
+			[
+				'methods' => 'GET',
+				'permission_callback' => function () {
+					return current_user_can( 'edit_posts' );
+				},
+				'callback' => $this->wrap_callback( function () {
+					return $this->index();
+				} ),
+			],
+		] );
+	}
+
+	private function index() {
+		$kit = Plugin::$instance->kits_manager->get_active_kit();
+
+		return new \WP_REST_Response( $kit->get_json_meta( static::META_KEY ) );
+	}
+
+
+	/**
+	 * PUT '/kit-elements-defaults/{type}'
+	 */
+	private function register_update() {
+		register_rest_route( static::NAMESPACE, static::BASE_URL . '/(?P<type>[\w\-_]+)', [
+			[
+				'methods' => 'PUT',
+				'args' => [
+					'type' => [
+						'type' => 'string',
+						'description' => 'The type of the element.',
+						'required' => true,
+						'validate_callback' => function( $param ) {
+							return $this->validate_type( $param );
+						},
+					],
+					'settings' => [
+						'description' => 'All the default values for the requested type',
+						'required' => true,
+						'type' => 'object',
+						'validate_callback' => function( $param ) {
+							return is_array( $param );
```

Comment:
The `type` will validate it, no need for the method

Reply:
For some reason it runs after the `sanitize_callback`, so I had to add a validation of my own